compiler/rustc_infer/src/traits/mod.rs

```diff
@@ -206,6 +206,50 @@ impl<'tcx> FulfillmentError<'tcx> {
     ) -> FulfillmentError<'tcx> {
         FulfillmentError { obligation, code, root_obligation }
     }
+
+    pub fn span(&self, tcx: TyCtxt<'tcx>) -> Span {
```

Comment:
I feel like it's incredibly easy to forget to use this function over `.obligation.cause.span` - can you consider another solution that doesn't give devs more choices that cause subtle inconsistencies when used differently?

Reply:
The issue about forgetting to call the method is fair, but do notice that we're already duplicating logic to modify that span after the fact in the current code. I'm concerned about properly tracking the span from inception if that will require additional memory or have non-trivial runtime impact on non-diagnostic codepaths. I'll see what options we have available to us that don't have those problems.

Reply:
I had to still use a closure for the deduplication logic (as it is very local to this one E0277 error for `Sized` locals), but otherwise the obligation span is now the one we want for all errors.